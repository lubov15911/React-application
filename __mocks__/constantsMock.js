export const moviePreviewData = {
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    vote_average: 7.8,
    tagline: 'The celebration of a lifetime',
    release_date: '2017-10-27',
    runtime: 105,
    genres: ["Adventure", "Comedy", "Family", "Animation"],
    overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz.',
};

export const filmsData = [{
    id: 354912,
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    release_date: '2017-10-27',
    genres: ["Adventure", "Comedy", "Family", "Animation"],
}, {
    id: 269149,
    poster_path: "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
    title: "Zootopia",
    release_date: "2016-02-11",
    genres: ["Animation", "Adventure", "Family", "Comedy"],
}, {
    id: 333339,
    poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
    title: 'Ready Player One',
    release_date: '2018-03-28',
    genres: ["Adventure", "Science Fiction", "Action"],
}, {
    id: 337167,
    poster_path: "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    title: "Fifty Shades Freed",
    release_date: "2018-03-28",
    genres: ["Drama", "Romance"],
}];

export const searchOptionsData = {
    title: 'Title',
    genre: 'Genre'
};

export const sortOptionsData = {
    rating: 'Rating',
    release: 'Release date'
};

export const searchValueData = 'Ready';
