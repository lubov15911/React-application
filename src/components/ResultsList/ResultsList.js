import React from 'react';
import PropTypes from 'prop-types';

import EmptyList from '../EmptyList/EmptyList';
import MoviesList from '../MoviesList/MoviesList';

import './ResultsList.scss';

const ResultsList = (props) => {
    const {
        totalResults,
        films,
        raiseClickEvent,
    } = props;

    return (
        <div className={`search-results ${!totalResults && 'no-items'}`}>
            {totalResults ?
                <MoviesList films={films} raiseClickEvent={raiseClickEvent} /> :
                <EmptyList />
            }
        </div>
    );
};
ResultsList.propTypes = {
    totalResults: PropTypes.number.isRequired,
    films: PropTypes.arrayOf(PropTypes.object).isRequired,
    raiseClickEvent: PropTypes.func.isRequired,
};

export default ResultsList;
