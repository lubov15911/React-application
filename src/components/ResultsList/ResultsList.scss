.search-results {
  padding: 6em;
  min-height: 55.5em;

  &.no-items {
    display: flex;
    justify-content: center;
    align-items: center;
  }
}
