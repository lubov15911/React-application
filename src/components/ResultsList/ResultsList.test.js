import React from 'react';
import { shallow } from 'enzyme';

import ResultsList from './ResultsList';

import { filmsData } from '../../../__mocks__/constantsMock';

describe('ResultsList', () => {
    const simpleFakeFunction = () => {};

    it('should render correctly films array', () => {
        const component = shallow(<ResultsList
            resultsAmount={filmsData.length}
            films={filmsData}
            raiseClickEvent={simpleFakeFunction}
        />);

        expect(component).toMatchSnapshot();
    });

    it('should render correctly empty films array', () => {
        const films = [] ;

        const component = shallow(<ResultsList
            resultsAmount={films.length}
            films={films}
            raiseClickEvent={simpleFakeFunction}
        />);

        expect(component).toMatchSnapshot();
    });
});
