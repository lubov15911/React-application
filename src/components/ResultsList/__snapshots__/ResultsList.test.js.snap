// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList should render correctly empty films array 1`] = `
<div
  className="search-results no-items"
>
  <EmptyList />
</div>
`;

exports[`ResultsList should render correctly films array 1`] = `
<div
  className="search-results false"
>
  <Connect(MoviesList) />
</div>
`;
